### PR TITLE
Enhance PAL initialization order

### PR DESCRIPTION
**Pull request by janvorli**
The CGroups initialization has _ASSERTE that is not asserting because
debugging support was being initialized after the CGroups initialization.

This change moves the debugging support initialization to the earliest
possible point. It also modifies it so that it doesn't depend on the
PAL env functions and uses getenv instead. And finally, initialization
of the critical section for debug printf is moved to the very end
of DBG_init_channels just as a little cleanup.

### CODE REVIEW

File: src/coreclr/src/pal/src/init/pal.cpp [hunk 1]
```diff
@@ -359,18 +359,34 @@ Initialize(
         gPID = getpid();
         gSID = getsid(gPID);
 
+        // Initialize the TLS lookaside cache
```
Comment (jkotas) [inline src/coreclr/src/pal/src/init/pal.cpp]: Nit: This comment is wrong. We got rid of the TLS lookaside cache a long time ago.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/src/pal/src/init/pal.cpp [hunk 2]
```diff
@@ -359,18 +359,34 @@ Initialize(
...
+        if (FALSE == TLSInitialize())
+        {
+            palError = ERROR_PALINIT_TLS;
+            goto done;
+        }
+
+        // Initialize debug channel settings before anything else.
+        // This depends on the environment, so it must come after
+        // EnvironInitialize.
```
Comment (jkotas) [inline src/coreclr/src/pal/src/init/pal.cpp]: EnvironInitialize is still way below. Is this comment right; or EnvironInitialize needs to be moved up as well?

Comment (janvorli) [reply src/coreclr/src/pal/src/init/pal.cpp]: Oops, forgotten to delete that portion of the comment. It is not needed as I've got rid of that dependency.

File: src/coreclr/src/pal/src/init/pal.cpp (hunk 1 above)
Comment (janvorli) [reply src/coreclr/src/pal/src/init/pal.cpp]: Fixed

File: src/coreclr/src/pal/src/init/pal.cpp (hunk 2 above)
Comment (janvorli) [reply src/coreclr/src/pal/src/init/pal.cpp]: Fixed

**Pull request merged**